### DISCUSSION TITLE
Upgrade always ran as per user not per machine

### DISCUSSION THREAD

**Discussion by DavidStrachanLexacom** [wixtoolset/issues#7242] [Questions]
I am fairly new to Wix and inherited an previous setup, so apologies in advance if my question is very basic.

The problem:
Upgrades are not removing previous install for some user.

Diagnostic:
I found that this was only impacting users that installed originally to the machine and not user. I setup and ran the logging and saw:  
`FindRelatedProducts: current install is per-user.  Related install for product '{GUID}' is per-machine.  Skipping...`

Question:
Why is an upgrade running install per-user when the original was set per-machine? 
Is this due to the upgrade not running Install, but running InstallNoShield (not elevating privileges)?
Where in the Wix Setup would the upgrade install be defined?

Additional information:
WE have MajorUpgrade set as:
`<MajorUpgrade DowngradeErrorMessage="A newer version of [ProductName] is already installed. Setup will now exit."/>`

**Comment by robmen** [comment]
Per-user/per-machine is a setting on the package. Your current package is per-user. The package already installed is per-machine. The Windows Installer will not upgrade across "install scopes".

So the real question is: Why does your package have a different install scope now?

**Reply by DavidStrachanLexacom** [reply]
That is my question correct, in my test I have bumped the version only. yet when I run the MSI it runs as per-user with no prompt asking which to run as (per-user or per-machine). Is there a setting that would say if upgrade then choose per-user?

Looking at the public path, we got from Welcome dialog, to licence dialog, to install scope dialog.  I have a control for install and installNoShield, but cannot see off hand how we determine which route to go

**Reply by robmen** [reply]
Well, something else changed too. WiX doesn't randomly select an install scope for your package every build. 😄

**Reply by DavidStrachanLexacom** [reply]
Again as per my original discussion stated:

> I am fairly new to Wix and inherited an previous setup, so apologies in advance if my question is very basic.

> Where in the Wix Setup would the upgrade install be defined?

My investigation into the issue has deemed that yes you are again correct "something else changed too", my question is as above what could of changed and where would be suggested to look.

I have a WixUI_Custom.wsx that is run and I am certain it will be in there, though cannot find anything that would indicate "this is the route we take in upgrade"

Attached is the wsx that might help
[wixui_customfile.txt](https://github.com/wixtoolset/issues/files/10815888/wixui_customfile.txt)

**Reply by barnson** [reply]
The intent appears to be to let the user choose between per-machine and per-user installs. That means the user would have to pick the same scope for an upgrade to work. That's one of the reasons letting the user choose is usually a bad idea...

**Reply by DavidStrachanLexacom** [reply]
Thanks, my issue is that on the upgrade though that choice is negated. The system does it for them but does it incorrectly.

**Reply by robmen** [reply]
Ahh, so that sounds like the bug. You'd probably benefit from reading the Windows Installer documentation about the [installation contexts](https://learn.microsoft.com/en-us/windows/win32/msi/installation-context) and the ALLUSERS property. A verbose log file will probably show you what is going on.

**Reply by DavidStrachanLexacom** [reply]
I have a verbose log, which showed:
`Action start 11:29:47: INSTALL.
MSI (c) (18:48) [11:29:47:062]: UI Sequence table 'InstallUISequence' is present and populated.
MSI (c) (18:48) [11:29:47:062]: Running UISequence
MSI (c) (18:48) [11:29:47:062]: PROPERTY CHANGE: Adding EXECUTEACTION property. Its value is 'INSTALL'.
MSI (c) (18:48) [11:29:47:062]: Doing action: FindRelatedProducts
MSI (c) (18:48) [11:29:47:062]: Note: 1: 2205 2:  3: ActionText 
Action 11:29:47: FindRelatedProducts. Searching for related applications
Action start 11:29:47: FindRelatedProducts.
MSI (c) (18:48) [11:29:47:062]: FindRelatedProducts: current install is per-user.  Related install for product '{463AC2BD-D0B9-4D51-925E-747EAFD14561}' is per-machine.  Skipping...
MSI (c) (18:48) [11:29:47:062]: FindRelatedProducts: current install is per-user.  Related install for product '{B37E5B31-15FF-43FA-820A-8C722C82869E}' is per-machine.  Skipping...
MSI (c) (18:48) [11:29:47:062]: FindRelatedProducts: current install is per-user.  Related install for product '{C737402B-1783-4661-85DE-5CFF23B97D17}' is per-machine.  Skipping...
MSI (c) (18:48) [11:29:47:062]: FindRelatedProducts: current install is per-user.  Related install for product '{70251809-51A0-4658-A9B3-9BE28663CF64}' is per-machine.  Skipping...
MSI (c) (18:48) [11:29:47:062]: FindRelatedProducts: current install is per-user.  Related install for product '{61F114C1-50E5-47F1-963D-06139D59AB14}' is per-machine.  Skipping...
MSI (c) (18:48) [11:29:47:063]: FindRelatedProducts: current install is per-user.  Related install for product '{87DAA7ED-F6EB-46E5-9BED-48ADD00204A7}' is per-machine.  Skipping...
MSI (c) (18:48) [11:29:47:063]: FindRelatedProducts: current install is per-user.  Related install for product '{3F528FDB-D073-4324-8D6E-D9336640D1E4}' is per-machine.  Skipping...
MSI (c) (18:48) [11:29:47:063]: FindRelatedProducts: current install is per-user.  Related install for product '{47C9AB7E-D119-40B5-A145-CE8032CF63CF}' is per-machine.  Skipping...
MSI (c) (18:48) [11:29:47:063]: FindRelatedProducts: current install is per-user.  Related install for product '{463AC2BD-D0B9-4D51-925E-747EAFD14561}' is per-machine.  Skipping...
MSI (c) (18:48) [11:29:47:063]: FindRelatedProducts: current install is per-user.  Related install for product '{B37E5B31-15FF-43FA-820A-8C722C82869E}' is per-machine.  Skipping...
MSI (c) (18:48) [11:29:47:063]: FindRelatedProducts: current install is per-user.  Related install for product '{C737402B-1783-4661-85DE-5CFF23B97D17}' is per-machine.  Skipping...
MSI (c) (18:48) [11:29:47:063]: FindRelatedProducts: current install is per-user.  Related install for product '{70251809-51A0-4658-A9B3-9BE28663CF64}' is per-machine.  Skipping...
MSI (c) (18:48) [11:29:47:063]: FindRelatedProducts: current install is per-user.  Related install for product '{61F114C1-50E5-47F1-963D-06139D59AB14}' is per-machine.  Skipping...
MSI (c) (18:48) [11:29:47:063]: FindRelatedProducts: current install is per-user.  Related install for product '{87DAA7ED-F6EB-46E5-9BED-48ADD00204A7}' is per-machine.  Skipping...
MSI (c) (18:48) [11:29:47:063]: FindRelatedProducts: current install is per-user.  Related install for product '{3F528FDB-D073-4324-8D6E-D9336640D1E4}' is per-machine.  Skipping...
MSI (c) (18:48) [11:29:47:063]: FindRelatedProducts: current install is per-user.  Related install for product '{47C9AB7E-D119-40B5-A145-CE8032CF63CF}' is per-machine.  Skipping...
Action ended 11:29:47: FindRelatedProducts. Return value 1.
MSI (c) (18:48) [11:29:47:063]: Doing action: PrepareDlg
MSI (c) (18:48) [11:29:47:063]: Note: 1: 2205 2:  3: ActionText `

What else would you look for?

**Reply by robmen** [reply]
As I mentioned above, the `ALLUSERS` property is important. Watch it in the case that works and does not work.
